core/iwasm/common/wasm_shared_memory.c

```diff
@@ -380,11 +387,17 @@ static bool
 map_remove_wait_info(HashMap *wait_map_, AtomicWaitInfo *wait_info,
                      void *address)
 {
+    os_mutex_lock(&wait_map_lock);
+    os_mutex_lock(&wait_info->wait_list_lock);
     if (wait_info->wait_list->len > 0) {
+        os_mutex_unlock(&wait_info->wait_list_lock);
+        os_mutex_unlock(&wait_map_lock);
         return false;
     }
+    os_mutex_unlock(&wait_info->wait_list_lock);
```

Comment:
Had better move this line to be after L399 `bh_hash_map_remove` to avoid other thread inserting `wait_node` into `wait_info->wait_list_lock` when it is to be removed?

Reply:
It can't happen because we insert only inside of `acquire_wait_info` and in the beginning of it we lock `wait_map_lock` which is as well locked here in L393.

A deletion won't happen as well because we already checked that the list is empty.

Reply:
OK, got it.